SConstruct

```diff
@@ -1178,257 +1156,136 @@ if env['python_package'] == 'new':
-                      "ignored.".format(v=major, option=env[py_pkg]))
-        else:  # pythonX_package is 'default'
-            # This dictionary has the default values for the Python related variables
-            default_py_vars = {'python{}_array_home': '', 'python{}_cmd': 'python{}'.format(major),
-                       'python{}_prefix': '$prefix'}
-
-            env[py_pkg] = env['python_package']
-            # Check whether any Python related variables are different from the default
-            for key, value in default_py_vars.items():
-                if env[key.format(major)] != value:
-                    print("WARNING: The value for {0} has been set and will be used instead "
-                          "of {1}".format(key.format(major), key.format('')))
-                else:
-                    env[key.format(major)] = env[key.format('')]
-
-        del env['python_version']
-
-# Make sure everything gets converted to properly versioned variables by deleting
-# these so they don't get used accidentally
-del env['python_package']
-del env['python_array_home']
-del env['python_cmd']
-del env['python_prefix']
-
-require_python = any([env['python{}_package'.format(p)] == 'full' for p in ['2', '3']])
-want_python = any([env['python{}_package'.format(p)] == 'default' for p in ['2', '3']])
-
-if require_python or want_python:
-    # Check for Cython:
-    try:
-        import Cython
-        cython_version = LooseVersion(Cython.__version__)
-        assert cython_version >= cython_min_version
-    except ImportError:
-        message = "Cython could not be imported by the Python interpreter running SCons."
-        have_cython = False
-    except AssertionError:
-        message = ("Cython is an incompatible version: "
-                   "Found {0} but {1} or newer is required.".format(cython_version,
-                                                                    cython_min_version))
-        have_cython = False
-    else:
-        have_cython = True
-    finally:
-        if not have_cython:
-            if require_python:
-                print('ERROR: ' + message)
-                sys.exit(1)
-            elif want_python:
-                print('WARNING: ' + message)
-                for py_pkg in ['python{}_package'.format(p) for p in ['2', '3']]:
-                    if env[py_pkg] == 'default':
-                        env[py_pkg] = 'minimal-default'
-        else:
-            print('INFO: Using Cython version {0}.'.format(cython_version))
+env['install_python_action'] = ''
+env['python_module_loc'] = ''
 
-def configure_minimal_python(py_ver):
-    # Test to see if we can run Python if the minimal interface is to be built
+if env['python_package'] in ('full', 'default'):
+    # Test to see if we can import numpy and Cython
     warn_no_python = False
     script = textwrap.dedent("""\
-        from __future__ import print_function
         import sys
         print('{v.major}.{v.minor}'.format(v=sys.version_info))
```

Comment:
It looks like this check for Python is duplicated when we check for the `minimal-default` case. Is there a way we can not have to do the check (that's just going to fail again) twice? Can we have a different bool for the case where Python itself can't be found, vs. NumPy or Cython not being found?